kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/templates/testclients/TestClientsJobTemplates.java

```diff
@@ -51,25 +56,173 @@ public static JobBuilder defaultAdminClientJob(String jobName, List<String> args
+     */
+    public static JobBuilder defaultKcatJob(String jobName, List<String> args) {
+        return baseClientJob(jobName)
+                .editSpec()
+                .editTemplate()
+                .editSpec()
+                .withContainers(new ContainerBuilder()
+                        .withName("kcat")
+                        .withImage(Constants.KCAT_CLIENT_IMAGE)
+                        .withImagePullPolicy(Constants.PULL_IMAGE_IF_NOT_PRESENT)
+                        .withArgs(args)
+                        .withSecurityContext(jobsSecurityContext())
                         .build())
-                .withRestartPolicy("Never")
                 .endSpec()
                 .endTemplate()
                 .endSpec();
     }
+
+    /**
+     * Default kafka go consumer job builder.
+     *
+     * @param jobName the job name
+     * @param args the args
+     * @return the job builder
+     */
+    public static JobBuilder defaultKafkaGoConsumerJob(String jobName, List<String> args) {
+        return baseClientJob(jobName)
+                .editSpec()
+                .editTemplate()
+                .editSpec()
+                .withContainers(new ContainerBuilder()
+                        .withName("kafka-go-consumer")
+                        .withImage(Constants.KAF_CLIENT_IMAGE)
+                        .withImagePullPolicy(Constants.PULL_IMAGE_IF_NOT_PRESENT)
+                        .withArgs(args)
+                        .withSecurityContext(jobsSecurityContext())
+                        .build())
+                .endSpec()
+                .endTemplate()
+                .endSpec();
+    }
+
+    private static EnvVar envVar(String name, String value) {
+        return new EnvVarBuilder()
+                .withName(name)
+                .withValue(value)
+                .build();
+    }
+
+    private static List<EnvVar> testClientsProducerEnvVars(String bootstrap, String topicName, int numOfMessages, String message) {
+        return List.of(
+                envVar(BOOTSTRAP_VAR, bootstrap),
+                envVar(DELAY_MS_VAR, "500"),
+                envVar(TOPIC_VAR, topicName),
+                envVar(MESSAGE_COUNT_VAR, String.valueOf(numOfMessages)),
+                envVar(MESSAGE_VAR, message),
+                envVar(PRODUCER_ACKS_VAR, "all"),
+                envVar(LOG_LEVEL_VAR, "INFO"),
+                envVar(CLIENT_TYPE_VAR, "KafkaProducer"));
+    }
+
+    private static List<EnvVar> testClientsConsumerEnvVars(String bootstrap, String topicName, int numOfMessages) {
+        return List.of(
+                envVar(BOOTSTRAP_VAR, bootstrap),
+                envVar(TOPIC_VAR, topicName),
+                envVar(MESSAGE_COUNT_VAR, String.valueOf(numOfMessages)),
+                envVar(GROUP_ID_VAR, "my-group"),
+                envVar(LOG_LEVEL_VAR, "INFO"),
+                envVar(CLIENT_TYPE_VAR, "KafkaConsumer"));
+    }
+
+    private static SecurityContext jobsSecurityContext() {
```

Comment:
IIRC you only need this if you create Pods directly.  As you are using Jobs, it shouldn't actually be required.   You should find the pod that results from your job has the SC defined for you.  Try it on OpenShift.  I suspect you can kill the method.